### PR TITLE
Add Cockroach DB Container Support

### PR DESCRIPTION
**Pull request by joelhess**
You have implemented a new container and would like to contribute it? Great! Here are the necessary steps:

- [x] You have added the new container as a module in the `testcontainers` directory (such as `testcontainers/my_fancy_container.py`).
- [x] You have added any new python dependencies in the `extras_require` section of `setup.py`.
- [x] You have added the `extra_requires` key to `requirements.in`.
- [x] You have updated all python requirements by running `make requirements` from the root directory.
- [x] You have added tests for the new container in the `tests` directory, e.g. `tests/test_my_fancy_container.py`.
- [x] You have added the name of the container (such as `my_fancy_container`) to the `test-components` matrix in `.github/workflows/main.yml` to ensure the tests are run.
- [x] You have rebased your development branch on `master` (or merged `master` into your development branch).

### CODE REVIEW

File: testcontainers/cockroachdb.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        driver="cockroachdb+psycopg2",
+        **kwargs
+    ):  # pylint: disable=too-many-arguments
+        super().__init__(image=image, **kwargs)
+        self.COCKROACH_USER = user or self.COCKROACH_USER
+        self.COCKROACH_PASSWORD = password or self.COCKROACH_PASSWORD
+        self.COCKROACH_DATABASE = dbname or self.COCKROACH_DATABASE
+        self.port_to_expose = self.DB_PORT
+        self.driver = driver
+        self.with_command("start-single-node --insecure")
```
Comment (mattupstate) [inline testcontainers/cockroachdb.py:59]: If you try to set a password via the `COCKROACH_PASSWORD` environment variable _and_ use the `--insecure` flag, you will get the following error:

```
....
CREATE DATABASE
finished creating default database "roach"
==== create key for new user noteable ====
ERROR: setting or updating a password is not supported in insecure mode
SQLSTATE: 28P01
Failed running "sql"
```

Comment (joelhess) [reply testcontainers/cockroachdb.py]: Thanks for that. I've updated the code, and added a test for that case.

**Comment by rafiss** [comment]
I'm an engineer from CockroachDB - this looks good to me!

**Comment by tillahoffmann** [comment]
> Please let me know if there's anything else that I need to do to get this merged. I'm trying to keep it up to date with Master at this point.

There a few more major-ish changes to prepare for v4.x (e.g. #296). I'd suggest not updating the PR until then (but thank you for putting the work in to keep it up to date). Shall I let you know when `master` is a bit more stable?

**Comment by joelhess** [comment]
> > Please let me know if there's anything else that I need to do to get this merged. I'm trying to keep it up to date with Master at this point.
> 
> There a few more major-ish changes to prepare for v4.x (e.g. #296). I'd suggest not updating the PR until then (but thank you for putting the work in to keep it up to date). Shall I let you know when `master` is a bit more stable?

Yes Please. Thanks for the follow up.

**Comment by joelhess** [comment]
@tillahoffmann Is the Main branch ready enough to update this to the new patterns?

**Comment by tillahoffmann** [comment]
Yes, fingers crossed.

**Pull request closed without merging**